pkg/loadbalancer/bpf_abi.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018-2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package loadbalancer
+
+// Use cgo to get the values of the service flags so that we can compare them
+// to the go version.
+
+// #define memcpy memcpy__dummy__
+// #define memmove memmove__dummy__
+// #define memset memset__dummy__
+// #define memcmp memcmp__dummy__
+// #include "bpf/ctx/xdp.h"
+// #undef memcpy
+// #undef memmove
+// #undef memset
+// #undef memcmp
+//
+// #include "lib/common.h"
+//
+// __u8 svc_flag_external_ip  = SVC_FLAG_EXTERNAL_IP_;
+// __u8 svc_flag_nodeport     = SVC_FLAG_NODEPORT_;
+// __u8 svc_flag_local_scope  = SVC_FLAG_LOCAL_SCOPE_;
+// __u8 svc_flag_hostport     = SVC_FLAG_HOSTPORT_;
+// __u8 svc_flag_affinity     = SVC_FLAG_AFFINITY_;
+// __u8 svc_flag_loadbalancer = SVC_FLAG_LOADBALANCER_;
+//
+// #cgo CFLAGS: -Wno-implicit-function-declaration -I../../bpf/ -I../../bpf/include
+import "C"
```

Comment:
This is using cgo which we preferably want to avoid. This seems to be only used for test code, so the easy fix would be to rename the file to `bpf_abi_test.go` so it only gets compiled for tests.

The other option, without using cgo for tests as well would be to somehow integrate it with the alignchecker which we already use to check struct alignment between bpf and Go code.

Reply:
Makes perfect sense.
Indeed, this was my first attempt, but tests do not seem to support cgo:
`can't load package: package github.com/cilium/cilium/pkg/loadbalancer: use of cgo in test /home/kkourt/isovalent/src/cilium/pkg/loadbalancer/loadbalancer_test.go not supported`

Reply:
Added #12436 as a first step towards using alignchecker.